### PR TITLE
Updated special characters for names and standardised the way ingredient is parsed

### PR DESCRIPTION
**Pull request by beatricetay**
- The names for ingredients and recipes can now be comprised of alphabets, numbers, white spaces, and special characters belonging to this set, { (, ), &, %, /, ', -, , } (Where ',' is one one of the special characters included)

- The way ingredients are parsed have been standardised to follow the format of [Quantity], [Name] for both user input and in the json file.

- I have tested this against @YuuTon9 's imported database, and have included that database in the doc/addressbook.json file :)

- Ingredients are now compulsory, a user needs to have at least one ingredient (from any of the 5 categories) in order to add a new recipe.

### CODE REVIEW

File: src/main/java/seedu/recipe/model/recipe/Name.java [hunk 1]
```diff
@@ -11,9 +11,13 @@
...
             "Names should only contain alphanumeric characters, spaces, or special characters from "
-                    + "this set {&, %, (, ), -}. The name should also not be blank.";
+                    + "this set {&, %, (, ), -, /}. The name should also not be blank.";
 
-    public static final String VALIDATION_REGEX = "^[\\sA-Za-z0-9\\()&%-]+$+";
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "^[\\sA-Za-z0-9\\()&%-/]+$+";
```
Comment (pekchinsiongbrian) [inline src/main/java/seedu/recipe/model/recipe/Name.java]: I think the validation regex here should be: "^[\sA-Za-z0-9'/\()&%-]+$+" (**note:** I added an apostrophe to the valid characters too)

When I tested the string: "Grandma's 100% mac & (string) cheese - for summer/winter" (without the quotes) it returned the whole string when the above regex was used (expected). However, when I tested this string against the validation regex in the line that I am commenting on, they could not find any matches!

Comment (beatricetay) [reply src/main/java/seedu/recipe/model/recipe/Name.java]: Thanks!

File: src/test/java/seedu/recipe/testutil/RecipeBuilder.java [hunk 2]
```diff
@@ -125,6 +129,11 @@ public RecipeBuilder withGrains(String ... grains) {
      * the {@code Recipe} that we are building.
      */
     public RecipeBuilder withVegetables(String ... vegetables) {
+        /*        if (isDefaultGrainPresent) {
```
Comment (pekchinsiongbrian) [inline src/test/java/seedu/recipe/testutil/RecipeBuilder.java]: Just to clarify, are we leaving these comments here or are they temporary? :)

**Review by pekchinsiongbrian — COMMENTED** [review]
wuhu thanks for the hard work Beatrice! :)

File: src/test/java/seedu/recipe/testutil/RecipeBuilder.java (hunk 2 above)
Comment (beatricetay) [reply src/test/java/seedu/recipe/testutil/RecipeBuilder.java]: hey! i commented those out as i was still trying to fix the failing test case for making the ingredients field compulsory.
Ive fixed it in the next commit and have uncommented these blocks of code :)

**Comment by vivienlee01** [comment]
thanks for the work Beats!

**Pull request merged**